.github/workflows/ci-model-regression.yml

```diff
@@ -0,0 +1,473 @@
+          restore-keys: ${{ runner.os }}-poetry-3.7
+
+      - name: Install Dependencies 📦
+        run: |
+          poetry install --extras full
+          make install
+          poetry run python -m spacy download de_core_news_md
+          poetry run python -m spacy link --force de_core_news_md de
+
+      - name: Validate that GPUs are working
+        run: |-
+          poetry run python -c 'from tensorflow.python.client import device_lib; print(device_lib.list_local_devices())' || true
+
+      - name: Run test
+        id: run_test
+        env:
+          TFHUB_CACHE_DIR: ~/.tfhub_cache/
+          OMP_NUM_THREADS: 1
+        run: |-
+          poetry run rasa --version
+
+          export NOW_TRAIN=$(gomplate -i '{{ (time.Now).Format time.RFC3339}}');
+          poetry run rasa train nlu -u dataset/$DATASET/train -c dataset/configs/$CONFIG --out models/$DATASET/$CONFIG
+          echo "::set-output name=train_run_time::$(gomplate -i '{{ $t := time.Parse time.RFC3339 (getenv "NOW_TRAIN") }}{{ (time.Since $t).Round (time.Second 1) }}')"
+
+          export NOW_TEST=$(gomplate -i '{{ (time.Now).Format time.RFC3339}}');
+          poetry run rasa test nlu --no-plot -u dataset/$DATASET/test -m models/$DATASET/$CONFIG --out results/$DATASET/$CONFIG
+
+          echo "::set-output name=test_run_time::$(gomplate -i '{{ $t := time.Parse time.RFC3339 (getenv "NOW_TEST") }}{{ (time.Since $t).Round (time.Second 1) }}')"
+          echo "::set-output name=total_run_time::$(gomplate -i '{{ $t := time.Parse time.RFC3339 (getenv "NOW_TRAIN") }}{{ (time.Since $t).Round (time.Second 1) }}')"
+
+      # Download the results of the previous runs
+      # The report file is extended with new results every next job run
+      - name: Download artifact
+        uses: actions/download-artifact@v2
+        continue-on-error: true
+        with:
+          name: report.json
+
+      - name: Generate a JSON file with a report / Publish results to Segment
+        env:
+          SUMMARY_FILE: "./report.json"
+          SEGMENT_TOKEN: ${{ secrets.SEGMENT_TOKEN }}
+          DATASET_NAME: ${{ matrix.dataset }}
+          RESULT_DIR: "${{ github.workspace }}/results"
+          CONFIG: ${{ matrix.config }}
+          TEST_RUN_TIME: ${{ steps.run_test.outputs.test_run_time }}
+          TRAIN_RUN_TIME: ${{ steps.run_test.outputs.train_run_time }}
+          TOTAL_RUN_TIME: ${{ steps.run_test.outputs.total_run_time }}
+        run: |-
+          export PR_URL="https://github.com/${GITHUB_REPOSITORY}/pull/${{ github.event.number }}"
+          poetry run pip install analytics-python
+          poetry run python .github/scripts/mr_publish_results.py
+          poetry run python .github/scripts/mr_generate_summary.py
+          cat $SUMMARY_FILE
+
+      - name: Upload an artifact with the report
+        uses: actions/upload-artifact@v2
+        with:
+          name: report.json
+          path: ./report.json
+
+  model_regression_test_cpu:
```

Comment:
why are we testing on CPU and GPU?

Reply:
I think it's the question to @dakshvar22, I guess we don't always need to run tests on GPU :D, @dakshvar22?

Reply:
Yeah, tests on some datasets can be fast enough on CPU as well, so a user may not need to run them on GPU. That's why we added a runner label to have the option to run the tests on CPU or GPU mode.